libs/ngrx-toolkit/src/lib/with-mutations.spec.ts

```diff
@@ -0,0 +1,537 @@
+  return of(value * 2).pipe(delay(delayInMsec));
+}
+
+function fail(_value: number, delayInMsec = 1000): Observable<number> {
+  return of(null).pipe(
+    delay(delayInMsec),
+    switchMap(() => throwError(() => ({ error: 'Test-Error' }))),
+  );
+}
+
+function createTestSetup(flatteningOperator = concatOp) {
+  function normalizeParam(param: Param): NormalizedParam {
+    if (typeof param === 'number') {
+      return {
+        value: param,
+        delay: 1000,
+        fail: false,
+      };
+    }
+
+    return {
+      value: param.value,
+      delay: param.delay ?? 1000,
+      fail: param.fail ?? false,
+    };
+  }
+
+  type SuccessParams = { result: number; params: Param };
+  type ErrorParams = { error: unknown; params: Param };
+
+  let onSuccessCalls = 0;
+  let onErrorCalls = 0;
+
+  let lastOnSuccessParams: SuccessParams | undefined = undefined;
+  let lastOnErrorParams: ErrorParams | undefined = undefined;
+
+  return TestBed.runInInjectionContext(() => {
+    const Store = signalStore(
+      withState({ counter: 3 }),
+      withMutations((store) => ({
+        increment: rxMutation({
+          operation: (param: Param) => {
+            const normalized = normalizeParam(param);
+
+            if (normalized.value instanceof Observable) {
+              return normalized.value.pipe(
+                switchMap((value) => {
+                  if (normalized.fail) {
+                    return fail(value, normalized.delay);
+                  }
+                  return calcDouble(value, normalized.delay);
+                }),
+              );
+            }
+
+            if (normalized.fail) {
+              return fail(normalized.value, normalized.delay);
+            }
+            return calcDouble(normalized.value, normalized.delay);
+          },
+          operator: flatteningOperator,
+          onSuccess: (result, params) => {
+            lastOnSuccessParams = { result, params };
+            onSuccessCalls++;
+            patchState(store, (state) => ({
+              counter: state.counter + result,
+            }));
+          },
+          onError: (error, params) => {
+            lastOnErrorParams = { error, params };
+            onErrorCalls++;
+          },
+        }),
+      })),
+    );
+
+    const store = new Store();
+    return {
+      store,
+      onSuccessCalls: () => onSuccessCalls,
+      onErrorCalls: () => onErrorCalls,
+      lastOnSuccessParams: () => lastOnSuccessParams,
+      lastOnErrorParams: () => lastOnErrorParams,
+    };
+  });
+}
+
+describe('withMutations with rxMutation', () => {
+  it('should update the state', fakeAsync(() => {
+    const testSetup = createTestSetup();
+    const store = testSetup.store;
+
+    expect(store.incrementStatus()).toEqual('idle');
+    expect(store.incrementProcessing()).toEqual(false);
+
+    store.increment(2);
+    expect(store.incrementStatus()).toEqual('processing');
+    expect(store.incrementProcessing()).toEqual(true);
+
+    tick(2000);
```

Comment:
A todo for the refactoring. We don't want tests that artificially extend the duration if it is not absolutely necesary.